http4k-security/digest/src/main/kotlin/org/http4k/security/digest/DigestEncoder.kt

```diff
@@ -0,0 +1,50 @@
+package org.http4k.security.digest
+
+import org.http4k.core.Method
+import org.http4k.security.Nonce
+import org.http4k.security.digest.Qop.Auth
+import org.http4k.security.digest.Qop.AuthInt
+import org.http4k.util.Hex
+import java.nio.charset.Charset
+import java.security.MessageDigest
+import kotlin.text.Charsets.ISO_8859_1
+
+class DigestEncoder(private val digester: MessageDigest, private val charset: Charset = ISO_8859_1) {
```

Comment:
Hah!  I had originally called it this, but ended up changing it to `DigestCalculator`.  I'm not opinionated on what we call it.